src/dune/cached_digest.ml

```diff
@@ -96,6 +96,10 @@ let refresh fn =
   set_with_stat fn digest stat;
   digest
 
+let refresh_and_chmod fn =
+  Path.chmod ~mode:0o222 ~op:`Remove fn;
+  refresh fn
```

Comment:
We are still doing the stat twice here.

Reply:
Mmmmh I lost some code it seems :thinking: